onnx/defs/tensor/defs.cc

```diff
@@ -330,16 +330,18 @@ ONNX_OPERATOR_SET_SCHEMA(
                 } else {
                   all_lengths_known = false;
                 }
-              } else if (shape.dim(j).has_dim_value()) {
-                if (output_shape->dim(j).has_dim_value()) {
-                  if (shape.dim(j).dim_value() !=
-                      output_shape->dim(j).dim_value()) {
+              } else {
+                const auto& output_dim = output_shape->dim(j);
+                const auto& input_dim = shape.dim(j);
+                if (!output_dim.has_dim_value() && !output_dim.has_dim_param())
+                  *output_shape->mutable_dim(j) = input_dim;
+                else
+                  if ((input_dim.has_dim_value() !=
+                       output_dim.has_dim_value()) ||
```

Comment:
I think it is possible for one of these to be a symbolic dimension and the other to have a value. I don't think we should throw a dimension-mismatch exception in that case.

Reply:
I think it would be useful to extract a utility function "mergeDim" (e.g. https://github.com/onnx/onnx/blob/fa568e4281f5008921e7b677ead901504cab491b/onnx/defs/shape_inference.h#L499 also tries to do the same thing) so that it can be used by multiple shape-inference methods

Reply:
Thanks for the suggestion. Done